pkg/machine/e2e/cp_test.go

```diff
@@ -0,0 +1,170 @@
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(125))
+		Expect(session.errorToString()).To(ContainSubstring("scp: dest open \"/etc/tmp.txt\": Permission denied"))
+
+		By("host directory to guest")
+		// copy contents into the guest
+		session, err = mb.setCmd(cp.withQuiet().withSrc(directoryPath).withDest(name + ":~/" + directory)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		// verify the content is in the guest
+		session, err = mb.setName(name).setCmd(sshMachine.withSSHCommand([]string{"ls", directory})).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+		Expect(session.outputToString()).To(Equal(fileInDirectory))
+
+		By("guest file to host")
+		// copy contents to the host
+		guestToHostFilePath := filepath.Join(GinkgoT().TempDir(), guestToHostFile)
+		session, err = mb.setCmd(cp.withQuiet().withSrc(name + ":~/" + file).withDest(guestToHostFilePath)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+		// check the contents are on the host
+		Expect(guestToHostFilePath).To(BeARegularFile())
+
+		if runtime.GOOS != "windows" {
+			// try to copy the file to a location on the host where permission will get denied
+			hostDirPath := filepath.Join(GinkgoT().TempDir(), "test-guest-copy-dir")
+			err = os.MkdirAll(hostDirPath, 0444)
+			Expect(err).ToNot(HaveOccurred())
+			hostFileInDirPath := filepath.Join(hostDirPath, file)
+			session, err = mb.setCmd(cp.withQuiet().withSrc(name + ":~/" + file).withDest(hostFileInDirPath)).run()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(session).To(Exit(125))
+			Expect(session.errorToString()).To(ContainSubstring(fmt.Sprintf("scp: open local \"%s\": Permission denied", hostFileInDirPath)))
+		}
+
+		By("guest directory to host")
+		// copy contents to the host
+		guestToHostDirPath := filepath.Join(GinkgoT().TempDir(), guestToHostDir)
+		session, err = mb.setCmd(cp.withQuiet().withSrc(name + ":~/" + directory).withDest(guestToHostDirPath)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+		// check the contents are on the host
+		Expect(filepath.Join(guestToHostDirPath, fileInDirectory)).To(BeARegularFile())
+
+		By("attempt copying file to a new directory")
+		// copy the file to a guest directory
+		session, err = mb.setCmd(cp.withQuiet().withSrc(filePath).withDest(name + ":~/directory/")).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(125))
+		Expect(session.errorToString()).To(ContainSubstring("scp: dest open \"directory/\": Failure"))
+
+		// try copying a guest file to a host directory
+		hostDirPath := filepath.Join(GinkgoT().TempDir(), "directory") + string(filepath.Separator)
+		session, err = mb.setCmd(cp.withQuiet().withSrc(name + ":~/" + file).withDest(hostDirPath)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(125))
+		switch runtime.GOOS {
+		case "windows":
+			hostDirPath = strings.ReplaceAll(hostDirPath, "\\", "/")
```

Comment:
there is filepath.ToSlash() which I think looks cleaner but this does the same thing so it is not a problem